### DISCUSSION TITLE
Customize Magit log/revision buffer naming behavior

### DISCUSSION THREAD

**Discussion by Thaodan** [magit/magit#5248] [Ideas]
# Description

When trying to understand changes over time inside commit it is useful to compare commits and/or histories.
Currently there's only one revision and/or log-buffer per repository.

# Alternative solutions considered

I tried to rename an existing log-buffer but that did not change that the log buffer opened was altered when I tried to look on another history of a repository.

# Proposed change

I think this feature could be added in multiple variations:
- An option to change the log buffer to be per range compared and rename the buffer to `magit-log: <repository>: <range>` as in e.g. `magit-log: magit: master`.
- Format string to name the buffer and reuse the buffer if the buffers naming matches the format modifiers used in the selected option for the new log to be opened

I think the first variant would be enough but some might want to customize the naming scheme of these buffers.
For revisions one buffer per revision in repository would be enough as an option from my point of view.

**Comment by kyleam** [comment]
> Currently there's only one revision and/or log-buffer per repository.

You can lock a buffer to prevent it from being reused (`(info "(magit)Modes and Buffers")`, <https://magit.vc/manual/magit.html#Modes-and-Buffers>).

**Comment by Thaodan** [comment]
That sounds like a very good option although a more transparent way of
e.g. viewing multiple revisions would be better.

**Reply by tarsius** [reply]
Too many suggestions are coming in right now, for me to handle them all.
For now try advising `magit-diff-setup-buffer` to always[^1] set `locked` to `t`.

[^1]: Or whenever you think that is appropriate.

**Comment by tarsius** [comment]
I've added a new option `magit-revision-use-dedicated-buffers`, but do not recommend enabling it (see its docstring).

See also https://github.com/magit/magit/discussions/5330#discussioncomment-12390660.
